src/taskpane/server.js

```diff
@@ -2364,16 +2820,182 @@ function createEntryFromRow(row, model, fieldTypeEnums, artifactIsHierarchical,
         // STANDARD FIELDS:
         // add standard fields in standard way - only add if field contains data
       } else if (value) {
-        // if the standard field is a multi select type as set in the switch above, pass the value through in an array
-        entry[fields[index].field] = (customType == "IntegerListValue") ? [value] : value;
+        if (isComment) {
+
+          if (fields[index].type == fieldTypeEnums.id) {
+            //for comments, we always have the ArtifactID
+            entry.ArtifactId = value;
+          }
+          else {//add the value normally
+            entry[fields[index].field] = (customType == "IntegerListValue") ? [value] : value;
+          }
+        }
+        else {
+          // if the standard field is a multi select type as set in the switch above, pass the value through in an array
+          entry[fields[index].field] = (customType == "IntegerListValue") ? [value] : value;
+        }
       }
     }
 
   }
-  return entry;
+  //double checking if the comment object has at least the necessary fields (i.e.: comment + ID). If not, add the correspondent fields
+  if (isComment) {
+    if (entry.ArtifactId && entry.Text) {
+      return entry
+    }
+    else if (entry.Text && sourceId) {
+      //we still need to get the artifactID
+      entry.ArtifactId = sourceId;
+      return entry
+    }
+    else return false;
+  }
+  else return entry;
 }
 
+// function creates a correctly formatted artifact object ready to send to Spira
+// it works through each field type to validate and parse the values so object is in correct form
+// any field that does not pass validation receives a null value
+// @param: row - a 'row' of data that contains a single object representing all fields
+// @param: model - full model with info about fields, dropdowns, users, etc
+// @param: fieldTypeEnums - object of all field types with enums
+// @param: artifactIsHierarchical - bool to tell function if this artifact has hierarchy (eg RQ and RL)
+// @param: lastIndentPosition - int used for calculating relative indents for hierarchical artifacts
+// @param: fieldsToFilter - enum used for selecting fields to not add to object - defaults to using all if omitted
+// @param: isUpdate - bool to flag id this is an update operation. If false, it is a creation operation
+// @param: isComment - bool to flag if we will return a comment entry (true) or custom/standard (false)
+function createAssociationEntryFromRow(row, model, fieldTypeEnums, originId) {

```

Comment:
It looks like the if blocks each have a lot of similar code. For example, below. Can the function be condensed so that only the bits that are different between each association type are in the if block? Or is better as is for reasons? If so, useful to add that reasoning as a comment in the code so future people know

var associationText = (row[index] + '').replace('.', ',');
        var associationIds = (associationText).split(',');

Reply:
Good catch! Due to the different structure of each association type, I originally thought that these blocks would be very different from each other - that obviously didn't happen. Already fixed in source code.